File: bin/create_directories.rb
# frozen_string_literal: true

def create_directories(prefix, listFile)
  list = File.readlines(listFile)
  list.each do |dir|
    dir = dir.chomp
    p "ensuring #{prefix}/#{dir} exists"
    system("mkdir -p #{prefix}/#{dir}")
  end
end

create_directories("#{Dir.home}/src/github.com/scanf", './DirectoriesList') if __FILE__ == $PROGRAM_NAME

File: bin/create_directories.ts
#!/usr/bin/env deno run --allow-read --allow-write --allow-env

import {
  ensureDirSync,
  readFileStrSync
} from "https://deno.land/std/fs/mod.ts";

export function createDirectories(prefix: string, listFile: string) {
  const directoriesFile = readFileStrSync(listFile);
  const files = directoriesFile.split("\n").map(x => `${prefix}/${x}`);
  for (const f of files) {
    console.log(`ensuring ${f} exists`);
    ensureDirSync(f);
  }
}

if (import.meta.main) {
  createDirectories(
    `${Deno.env().HOME}/src/github.com/scanf`,
    "./DirectoriesList"
  );
}
